Build/build.requirements.psd1

```diff
@@ -1,19 +1,51 @@
 ﻿@{
     # Some defaults for all dependencies
-    PSDependOptions = @{
-        Target = '$ENV:USERPROFILE\Documents\WindowsPowerShell\Modules'
+    PSDependOptions    = @{
+        Target    = '$ENV:USERPROFILE\Documents\WindowsPowerShell\Modules'
         AddToPath = $True
-        Parameters = @{
+        <# Parameters = @{
             Force = $True
-        }
+        } #>
     }
 
     # Grab some modules without depending on PowerShellGet
-    'psake' = @{ DependencyType = 'PSGalleryNuget' }
-    'PSDeploy' = @{ DependencyType = 'PSGalleryNuget' }
-    'BuildHelpers' = @{ DependencyType = 'PSGalleryNuget' }
-    'Pester' = @{
+    'psake'            = @{
+        DependencyType = 'PSGalleryNuget'
+        Force = $True
+    }
+    'PSDeploy'         = @{
+        DependencyType = 'PSGalleryNuget'
+        Force = $True
+    }
+    'BuildHelpers'     = @{
+        DependencyType = 'PSGalleryNuget'
+        Force = $True
+    }
+    'Pester'           = @{
         DependencyType = 'PSGalleryNuget'
-        Version = '3.4.6'
+        Version        = '3.4.6'
+        Force = $True
+    }
+    # Module dependencies for Azure-related deployment types
+    'Az.Automation'    = @{
+        DependencyType = 'PSGalleryNuget'
+        Force = $True
+        DependsOn      = 'UninstallAzureRM'
+    }
+    'Az.Storage'       = @{
+        DependencyType = 'PSGalleryNuget'
+        Force = $True
+        DependsOn      = 'UninstallAzureRM'
+    }
+
+    'UninstallAzureRm' = @{
```

Comment:
So!  Clean build servers are ideal, but if someone were to do some local building, this would be a bit of a problem - is there a reason to cleanout AzureRM*?  If so I think we'd want to gate this by build system (e.g. appveyor) or have some way to flag that it's okay to do it, and not do it by default?

Reply:
Hi Warren! Thanks for bringing my attention to that!

Yeah, the build image defined in the AppVeyour pipeline contains preinstalled Microsoft Azure PowerShell 5.7 (https://www.appveyor.com/docs/windows-images-software/#azure), which is an outdated AzureRM module.
As I'm using the Az modules in the new deployment types, I cannot install the Az modules without removing the AzureRM modules from a build environment first (https://ci.appveyor.com/project/RamblingCookieMonster/psdeploy/builds/32949255).

I've updated the dependency handling only to uninstall the AzureRM modules when running in AppVeyour so that not to intrude in the environment when building locally.

P.S. Also, I've updated the ARM deployment type to use the Az modules instead of AzureRM.

Reply:
Please, let me know if there are any other concerns.
As I consider myself a novice in PowerShell, I really appreciate your feedback. ✌

Reply:
@RamblingCookieMonster Are you okay with the proposed workaround?

Reply:
I think so.  If someone is testing this in some janky CI environment where they set `$ENV:AppVeyor` to a string 'false' and it triggers... hopefully they pin versions and notice this : )  It seems so unlikely we can merge and those folks could open a PR, if they exist!